### DISCUSSION TITLE
Trouble with reading FunctionArrayAux variables into ConservativeAdvection kernel.

### DISCUSSION THREAD

**Discussion by danielwardstfc** [idaholab/moose#25366] [Q&A General] [answered]
Hi,

I have read some velocity data from OpenFOAM into MOOSE using FunctionArrayAux and the answer to my previous [question](https://github.com/idaholab/moose/discussions/24784#discussioncomment-6874784).

I am now trying to read this into the ConservativeAdvection kernel so that I can advect the temperature using my velocity field, but I think I have run into issues with variable types.  The error message I'm receiving is as below:

```
 *** ERROR ***
The named variable is an array variable, try a 'coupledArray[Value/Gradient/Dot/etc]...' function instead
```

As far as I can tell, the variables of ArrayAux kernel are of type RealEigenVector, whereas the variables required by ConservativeAdvection are RealVectorValue.  Do you know if I can easily convert from one to the other?  Or is this a case of rewriting the ConservativeAdvection kernel?

Best,

Daniel

P.S. My input file is as follows:

```
[Mesh]
  file = in/cylinder.msh
[]

[AuxVariables]
  [velocity]
    family = LAGRANGE
    order = FIRST
    components = 3
  []
[]

[AuxKernels]
  [velocity-ker]
    type = FunctionArrayAux
    variable = velocity
    functions = 'uvelocity vvelocity wvelocity'
  []
[]

[Variables]
  [T]
    order = FIRST
    family = LAGRANGE
  []
[]

[Kernels]
  [conservative-advection]
    type = ConservativeAdvection
    variable = T
#    velocity = '0 0 1'
    velocity = 'velocity'
    upwinding_type = full
  []
[]


[BCs]
  [./bottom]
    type = DirichletBC
    variable = T
    boundary = 'inlet'
    value = 1
  [../]
  [./top]
    type = DirichletBC
    variable = T
    boundary = 'outlet'
    value = 0
  [../]
[]

[Executioner]
  type = Steady
  solve_type = 'PJFNK'
[]

[Outputs]
  execute_on = 'timestep_end'
  exodus = true
[]

[UserObjects]
  [reader_nearest]
    type = PropertyReadFile
    prop_file_name = 'in/velocity_field_edit.csv'
    read_type = 'voronoi'
    nprop = 6
    nvoronoi = 40291
  []
[]

[Functions]
  [uvelocity]
    type = PiecewiseConstantFromCSV
    read_prop_user_object = 'reader_nearest'
    read_type = 'voronoi'
    column_number = '3'
  []
  [vvelocity]
    type = PiecewiseConstantFromCSV
    read_prop_user_object = 'reader_nearest'
    read_type = 'voronoi'
    column_number = '4'
  []
  [wvelocity]
    type = PiecewiseConstantFromCSV
    read_prop_user_object = 'reader_nearest'
    read_type = 'voronoi'
    column_number = '5'
  []
[]
```

**Comment by lindsayad** [comment, marked as answer]
If you wanted to work within the framework of existing objects I'd suggest using `ArrayVariableComponent` to get scalar field aux variables and then `ParsedVectorAux` to convert the scalar field components into a vector variable.

Alternatively you could write your own kernel or make a modification to `ConservativeAdvection`
